onnxruntime/core/providers/cpu/nn/pool.cc

```diff
@@ -224,6 +224,67 @@ Status MaxPoolV8::ComputeImpl(OpKernelContext* context) const {
   return Status::OK();
 }
 
+template <typename T>
+Status MaxPoolV8::ComputeImplOptimized(OpKernelContext* context) const {
+
+  const auto* X = context->Input<Tensor>(0);
+  const auto* X_data = X->template Data<T>();
+  const TensorShape& x_shape = X->Shape();
+  std::vector<int64_t> pads = pool_attrs_.pads;
+  std::vector<int64_t> output_dims = pool_attrs_.SetOutputSize(x_shape, x_shape[1], &pads);
+  const auto& dilations = pool_attrs_.dilations;
+  const auto& pool_size = pool_attrs_.kernel_shape;
+
+  if (dilations[0] != stride_h() ||
+      pool_size.size() > 1 && dilations[1] != stride_w() ||
+      pool_size.size() > 2 && dilations[2] != stride_d() ||
+      context->Output(1, output_dims)) {
+
+    return Status(common::ONNXRUNTIME, common::FAIL);
```

Comment:
error message along with the fail status ?

Reply:
The returned status will not be reported or consumed anywhere else beyond its caller.

Reply:
The error msg should trickle all the way up the chain. Is that not the case?

Reply:
No it is not, let me return a bool to relieve the confusion.

Reply:
Let's fix the error msg propagation then. How would a user know if something went wrong.

Reply:
Done.